tools/cross/format.py

```diff
@@ -74,17 +83,30 @@ def check_clang_format() -> bool:
         exit(1)
 
 
-def filter_files_by_exts(
-    files: List[str], dir_path: str, exts: Tuple[str, ...]
-) -> List[str]:
+def filter_files_by_globs(
+    files: List[Path], dir_path: Path, globs: Tuple[str, ...]
+) -> List[Path]:
     return [
         file
         for file in files
-        if (dir_path == "." or file.startswith(dir_path + "/")) and file.endswith(exts)
+        if is_relative_to(file, dir_path) and matches_any_glob(globs, file)
     ]
 
 
-def clang_format(files: List[str], check: bool = False) -> bool:
+def matches_any_glob(globs: Tuple[str, ...], file: Path) -> bool:
+    return any(file.match(glob) for glob in globs)
+
+
+def is_relative_to(path: Path, prefix: Path) -> bool:
+    # Polyfill for method introduced in Python 3.9
```

Comment:
Is anyone on a system still using python 3.9?

Reply:
@danlapid suggested we keep 3.8 compatibility as it isn't too hard to do

Reply:
No strong feelings on my end, I don't hate the polyfill but I don't care if we want to update the minimum required to any version.

Reply:
OK let's go with 3.9. I'm just itching to remove those imports from `typing`